### PR TITLE
Stop multiple or all scans at once.

### PR DESCRIPTION
**Pull request by elyousfi5**
This PR enhances the `scan stop` command to allow stopping multiple scans or all scans at once, improving the functionality from only stopping a single scan previously. The new feature provides more flexibility and efficiency in managing scans via the CLI.

#### **Changes Introduced**
1. **New Features**:
   - **Stop Multiple Scans**: Users can now provide multiple scan IDs to stop them in a single command:
     ```bash
     ostorlab scan --runtime=local stop 1 2 3
     ```
   - **Stop All Scans**: Added `--all` flag to stop all running scans:
     ```bash
     ostorlab scan --runtime=local stop --all
     ```

2. **Error Handling**:
   - If neither scan IDs nor the `--all` flag is provided, the command raises a `UsageError`.

3. **Improvements**:
   - Clear and actionable console messages:
     - Number of scans being stopped is displayed.
     - Warning is shown if `--all` is used but no scans are running.

4. **Backward Compatibility**:
   - The command still supports stopping a single scan as before.

#### **Testing**
Added new unit tests to ensure the functionality works as expected:
1. **Stopping multiple scans**: Verified that the `stop` method is called for each provided scan ID.
2. **Stopping all scans**: Validated that all running scans are stopped when `--all` is used.
3. **No scans running**: Confirmed appropriate warning message is displayed when `--all` is used but no scans are found.

### CODE REVIEW

File: src/ostorlab/cli/scan/stop/stop.py [hunk 1]
```diff
@@ -11,14 +13,35 @@
 
 
 @scan.command()
-@click.argument("scan_id", required=True)
+@click.argument("scan_ids", nargs=-1, type=int, required=False)
+@click.option(
+    "--all",
```
Comment (3asm) [inline src/ostorlab/cli/scan/stop/stop.py:18]: -a

File: src/ostorlab/cli/scan/stop/stop.py [hunk 2]
```diff
@@ -11,14 +13,35 @@
...
+    "stop_all",
+    is_flag=True,
+    help="Stop all running scans",
+    default=False,
+)
 @click.pass_context
-def stop(ctx: click.core.Context, scan_id: int) -> None:
-    """Stop a scan.\n
+def stop(ctx: click.core.Context, scan_ids: Tuple[int, ...], stop_all: bool) -> None:
+    """Stop one, multiple scans.\n
```
Comment (amine3) [inline src/ostorlab/cli/scan/stop/stop.py:26]: Stop one or multiple scans

File: src/ostorlab/cli/scan/stop/stop.py [hunk 3]
```diff
@@ -11,14 +13,35 @@
...
+    if len(scan_ids) == 0 and stop_all is False:
+        raise click.UsageError("Either provide scan IDs or use --all flag")
 
     runtime_instance = ctx.obj["runtime"]
-    with console.status("Stopping scan"):
+    if stop_all is True:
+        scans_list = runtime_instance.list()
+        ids_to_stop = [s.id for s in scans_list]
+        if len(ids_to_stop) == 0:
+            console.warning("No running scans found")
```
Comment (3asm) [inline src/ostorlab/cli/scan/stop/stop.py:40]:
```suggestion
            console.warning("No running scans found.")
```

File: src/ostorlab/cli/scan/stop/stop.py [hunk 4]
```diff
@@ -11,14 +13,35 @@
...
+            return
+    else:
+        ids_to_stop = list(scan_ids)
+
+    console.info(f"Stopping {len(ids_to_stop)} scan(s)")
```
Comment (3asm) [inline src/ostorlab/cli/scan/stop/stop.py:45]:
```suggestion
    console.info(f"Stopping {len(ids_to_stop)} scan(s).")
```

**Review by 3asm — DISMISSED** [review]

**Review by amine3 — CHANGES_REQUESTED** [review]

**Pull request merged**